### DISCUSSION TITLE
Conversion from old config to new theme

### DISCUSSION THREAD

**Discussion by durdo** [davatorium/rofi#1393] [Theming] [answered]
Hello!

First of all, I want to thank all the contributors to this project, I very much like rofi and I appreciate the work being done here.

I have an old rasi configuration where I specified my theme (using width,  lines, columns, color-normal, so on and so forth) after a recent update I saw that there is a new config for theming.

_My question is_: **Is there a trivial way to convert the old config to the new one?**

I am very satisfied with the look I had but, at first try, I had difficulties trying to translate to the new format. I really wouldn't like to have to learn a new css-like configuration only to reproduce something that I already have.

Thank you,
Eduardo

**Comment by DaveDavenport** [comment, marked as answer]
The previous versions had an one-stop -upgrade-config option and gave a warning on each launch when an old version was launched.. 
Before that (since 2017) it could generate a new config file from your old config file.

The dependency on xcb-xrm was dropped and the old format cannot be parsed anymore.

**Reply by durdo** [reply]
Thank you for the prompt reply!

The version I had before was 1.6.1, I rolled back and there is no warning. 

And running -upgrade-config results in:

```bash
(process:746655): Rofi-WARNING **: 12:59:06.366: New configuration file already exists: /home/eduardo/.config/rofi/config.rasi
```

What am I missing here?

**Reply by DaveDavenport** [reply]
you already have a new configuration file, it will not overwrite that, as it might causing in loss of data.

**Reply by durdo** [reply]
I understand, so my question remains:

This config.rasi that I have is exactly the one that have my before mentioned "theme" configuration. Is there a trivial way to move it to the new theming scheme?

**Reply by DaveDavenport** [reply]
@durdo try the following:
```bash
rofi -dump-config > newconfig.rasi
rofi -dump-theme  >> newconfig.rasi
```
You can test this with  `rofi -config newconfig.rasi -show run`

then copy newconfig.rasi to ~/.config/rofi/config.rasi

**Reply by DaveDavenport** [reply]
> I understand, so my question remains:
> 
> This config.rasi that I have is exactly the one that have my before mentioned "theme" configuration. Is there a trivial way to move it to the new theming scheme?

I don't know what is going wrong or what is failing, so it is hard to answer this question and I need to guess.

**Reply by DaveDavenport** [reply]
The 'color-normal', color-urgent options have been deprecated for many many years now.
As mentioned in the releasenotes for maintainability we cannot keep supporting formats deprecated many years ago.
I think the above two steps solution might still work.

**Reply by durdo** [reply]
I followed your instruction and had partial success!

The colors were translated perfectly, but some configs got lost: width, columns and lines (not sure if anything else ).

> The 'color-normal', color-urgent options have been deprecated for many many years now.

I understand. I spent some time in the past to get the look that I liked and never went back to update it. Doing this now because I am being forced to.

Thank you for all the help.

**Reply by DaveDavenport** [reply]
append to your config.rasi:

```css
listview {
lines: 4;
columns: 3;
}

window {
  width: 50%;
}
```
with the values you want.

**Reply by DaveDavenport** [reply]
> I followed your instruction and had partial success!
> 
> The colors were translated perfectly, but some configs got lost: width, columns and lines (not sure if anything else ).
> 
> > The 'color-normal', color-urgent options have been deprecated for many many years now.
> 
> I understand. I spent some time in the past to get the look that I liked and never went back to update it. Doing this now because I am being forced to.
> 
> Thank you for all the help.

I understand, but regrettable I cannot keep things around forever giving limited development time.

**Reply by durdo** [reply]
I am very sorry for wasting your time helping a "late adopter", I'll try to keep up-to-date to avoid this kind of problem in the future.

> listview {
> lines: 4;
> columns: 3;
> }
>
> window {
>   width: 50%;
> }

This did the trick. Now everything is working, no warnings, and the theme was correctly ported!

> I understand, but regrettable I cannot keep things around forever giving limited development time.

Yeah, completely understandable. It would be unreasonable to ask for full retro compatibility _forever_.

Thank you for your patience,
Keep up the great work!

![rofi](https://user-images.githubusercontent.com/18620988/131375500-5286d2b8-1a34-4a41-bba4-ecde5e0c4aab.png)

**Reply by DaveDavenport** [reply]
happy to hear it works.

**Reply by DaveDavenport** [reply]
> I am very sorry for wasting your time helping a "late adopter", I'll try to keep up-to-date to avoid this kind of problem in the future.

If it goes friendly like this, I don't consider it a waste of time. Happy to help.

**Reply by DaveDavenport** [reply]
(if the question is answered, can you mark it? thx)
